aiqc/__init__.py

```diff
@@ -4522,7 +4518,10 @@ def colIndices_from_colNames(column_names:list, desired_cols:list):
 	def cols_by_indices(arr:object, col_indices:list):
 		# Input and output 2D array. Fetches a subset of columns using their indices.
 		# In the future if this needs to be adjusted to handle 3D array `[:,col_indices,:]`.
-		subset_arr = arr[:,col_indices]
+		if arr.ndim == 1:
```

Comment:
Did you run into any trouble with 1D arrays? I ask because figuring out the required dimensions during fit/transform is a pain.

Reply:
I am surprised to see 1D arrays here. especially since you returned the array as is. it should probably error if anything. did you have to use `cols_by_indices` with 1D data?

Reply:
this function was erroring out with 1D arrays because the sub_arr was being indexed along 2nd dimension which doesn't exist for 1D arrays, the 1D arrays are a quirk of the text dataset API: to_numpy() (/to_pandas()) returns a 1D list of strings that are fit/transformed using the sklearn preprocessor, can change the output there to be a 2d array